File: 1538-maximum-points-you-can-obtain-from-cards/1538-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        int lsum = 0;
        int rsum = 0;

        for(int i=0; i<k; i++){
            lsum += cardPoints[i];
        }
        int r = n-1;
        int maxsum = lsum;
        for(int i = k-1; i >=0; i--){
            lsum -= cardPoints[i];
            rsum += cardPoints[r];
            r--;
            maxsum = max(maxsum, lsum+rsum);
        }
        return maxsum;
    }   
};

File: 1538-maximum-points-you-can-obtain-from-cards/1538-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        if len(cardPoints) == k:
            return sum(cardPoints)
        n = len(cardPoints)
        lsum = 0
        rsum = 0
        for i in range(k):
            lsum += cardPoints[i]

        r = n-1 
        l = k-1
        maxsum = lsum

        while l >= 0:
            lsum -= cardPoints[l]
            rsum += cardPoints[r]
            maxsum = max(maxsum, lsum + rsum)
            l-=1
            r-=1
        
        return maxsum